homeassistant/components/lifx/coordinator.py

```diff
@@ -156,3 +202,43 @@ async def async_set_color_zones(
                 apply=apply,
             )
         )
+
+    async def async_update_wifi_info(self) -> None:
+        """Get the bulb's wifi signal strength."""
+        _LOGGER.debug("Updating wifi signal strength for %s", self.device.label)
+        try:
+            wifi_info = await async_execute_lifx(self.device.get_wifiinfo)
+            self._rssi = int(math.floor(10 * math.log10(wifi_info.signal) + 0.5))
+        except asyncio.TimeoutError as ex:
+            raise UpdateFailed(
+                f"Failed to fetch RSSI sensor data from device: {self.device.ip_addr}"
+            ) from ex
+
+    async def async_update_hev_cycle(self) -> None:
+        """Get the current and last HEV cycle data from the bulb."""
+        _LOGGER.debug("Updating HEV cycle sensor values for %s", self.device.label)
+        try:
+            hev_cycle: StateHevCycle = await async_execute_lifx(
+                self.device.get_hev_cycle
+            )
+            self._hev_cycle = {
+                HEV_CYCLE_DURATION: hev_cycle.duration,
+                HEV_CYCLE_REMAINING: hev_cycle.remaining,
+                HEV_CYCLE_LAST_POWER: hev_cycle.last_power,
+            }
+
+            last_result: StateLastHevCycleResult = await async_execute_lifx(
+                self.device.get_last_hev_cycle_result
+            )
+            self._hev_cycle[HEV_CYCLE_LAST_RESULT] = (
+                str(last_result.result_str)
+                .title()
+                .replace("_", " ")
+                .replace("Homekit", "HomeKit")
+                .replace("Lan", "LAN")
+            )
+            _LOGGER.debug(self._hev_cycle)
```

Comment:
Please move code out of the coordinator that won't raise.
`
            hev_cycle: StateHevCycle = await async_execute_lifx(
                self.device.get_hev_cycle
            )
            last_result: StateLastHevCycleResult = await async_execute_lifx(
                self.device.get_last_hev_cycle_result
            )
`

Then process it in an `else`